src/lib/storage/table.cpp

```diff
@@ -23,6 +23,11 @@ Table::Table(const ChunkOffset target_chunk_size) : _max_chunk_size{target_chunk
   _chunks.push_back(std::make_shared<Chunk>());
 }
 
+void Table::add_column_definition(const std::string& name, const std::string& type) {
+  _col_names.push_back(name);
```

Comment:
Could add a check whether the table is non empty before adding column definitions.